music-assistant: init at 2.0.7

mweinelt opened the pull request:
## Description of changes

https://github.com/music-assistant/server
https://github.com/music-assistant/frontend

and a few other things from https://github.com/music-assistant

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mweinelt committed:
python311Packages.memory-tempfile: init at 2.2.3

mweinelt committed:
music-assistant: init at 2.0.7

mweinelt committed:
python311Packages.py-opensonic: init at 5.1.1

 A python library to wrap the Open Subsonic REST API.

mweinelt committed:
nixos/music-assistant: init

mweinelt committed:
nixos/tests/music-assistant: init

mweinelt commented on nixos/modules/services/audio/music-assistant.nix:
```diff
@@ -0,0 +1,113 @@
+      type = listOf (enum cfg.package.providerNames);
+      default = [];
+      example = [
+        "opensubsoic"
```
```suggestion
        "opensubsonic"
```

SuperSandro2000 commented on pkgs/development/python-modules/py-opensonic/default.nix:
```diff
@@ -0,0 +1,43 @@
+
+  build-system = [
+    setuptools
+    wheel
```
Why do we need wheel here? The package is not referring to it except through GHA.

SuperSandro2000 commented on nixos/modules/services/audio/music-assistant.nix:
```diff
@@ -0,0 +1,113 @@
+
+    extraOptions = mkOption {
+      type = listOf str;
+      default = [];
```
We probably want to pass by default --config to not write the data into /var/lib/music-assistant/.musicassistant

```suggestion
      default = [ "--config" "/var/lib/music-assistant" ];
```

mweinelt commented:
- Scrolling through albums/artists/titles is super slow
  - ```
    ERROR (MainThread) [music_assistant.helpers.process] Detected that you are running the (docker) container without permissive access rights. This will impact performance !
    ```
- They have no browser playback :(

mweinelt committed:
nixos/doc/rl-2411: add music-assistant

mweinelt committed:
buildHomeAssistantComponent: support custom sentences

Custom components can also ship with sentences that extend the grammar-
based voice infrastructur of Home Assistant.

mweinelt committed:
home-assistant-custom-components.mass: init at 2024.6.2

SuperSandro2000 commented on pkgs/by-name/mu/music-assistant/package.nix:
```diff
@@ -0,0 +1,119 @@
+
+  meta = with lib; {
+    changelog = "https://github.com/music-assistant/server/releases/tag/${version}";
+    description = "Music Assistant is a music library manager for your offline and online music sources which can easily stream your favourite music to a wide range of supported players";
```
```suggestion
    description = "Music Assistant is a music library manager for your music which can stream it to a wide range of supported players";
```
I tried to shorten that a bit
